### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by kay0u**
- #712 
- #711

### CODE REVIEW

**Comment by kay0u** [comment]
!testme

File: scripts/config [hunk 1]
```diff
@@ -105,6 +105,24 @@ set__fpm_free_footprint() {
...
+    ynh_print_info "Set permissions, it may take some time..."
+    chown -R $app:www-data "$install_dir"
+    chown -R $app: "$data_dir"
+    find $install_dir/ -type f -print0 | xargs -r0 chmod 0644
+    find $install_dir/ -type d -print0 | xargs -r0 chmod 0755
+    find $data_dir/data/ -type f -print0 | xargs -r0 chmod 0640
+    find $data_dir/data/ -type d -print0 | xargs -r0 chmod 0750
+    chmod 640 "$install_dir/config/config.php"
+    chmod 755 /home/yunohost.app
+    chmod 750 $install_dir
```
Comment (kay0u) [inline scripts/config:123]: We can also run it in a sub-process to not take the yunohost lock just to reset the permissions on these files

**Comment by Thovi98** [comment]
I’ve just made the upgrade to this testing branch. Everything went fine, thank you for the good work!

The Nextcloud checkup still warns about `Votre serveur web n’est pas configuré correctement pour résoudre les URL .well-known, a échoué sur : /.well-known/caldav Pour plus d’information, voir la [documentation ↗](https://docs.nextcloud.com/server/29/go.php?to=admin-setup-well-known-URL).`.  The [upgrade logs](https://paste.yunohost.org/raw/uvezoxalih) say this:
```
grep --quiet '# configuration file /etc/nginx/conf.d/cloud.maindomain.tld.d/nextcloud.conf:'
2024-08-01 16:29:04,615: DEBUG - + ynh_exec_warn_less nginx -T
2024-08-01 16:29:04,616: DEBUG - + [[ 2 -eq 1 ]]
2024-08-01 16:29:04,618: DEBUG - + nginx -T
2024-08-01 16:29:05,617: DEBUG - + break
2024-08-01 16:29:05,617: DEBUG - + is_url_handled --domain=cloud.maindomain.tld --path=/.well-known/caldav
2024-08-01 16:29:05,619: DEBUG - + declare -Ar args_array
2024-08-01 16:29:05,774: DEBUG - ++ curl --insecure --silent --output /dev/null --write-out '%{http_code};%{redirect_url}' https://127.0.0.1/.well-known/caldav --header 'Host: cloud.maindomain.tld' --resolve cloud.maindomain.tld:443:127.0.0.1
2024-08-01 16:29:06,107: DEBUG - + local 'curl_output=301;https://cloud.maindomain.tld/index.php/.well-known/caldav'
2024-08-01 16:29:06,108: DEBUG - + local http_code=301
2024-08-01 16:29:06,110: DEBUG - + local redirection=https://cloud.maindomain.tld/index.php/.well-known/caldav
2024-08-01 16:29:06,111: DEBUG - + '[' 301 = 404 ']'
2024-08-01 16:29:06,111: DEBUG - + [[ https://cloud.maindomain.tld/index.php/.well-known/caldav =~ /yunohost/admin ]]
2024-08-01 16:29:06,112: DEBUG - + [[ https://cloud.maindomain.tld/index.php/.well-known/caldav =~ /yunohost/sso ]]
2024-08-01 16:29:06,112: DEBUG - + ynh_print_warn '--message=Another app already uses the domain cloud.maindomain.tld to serve a CalDAV/CardDAV feature. You may encounter issues when dealing with your calendar or address book.'
2024-08-01 16:29:06,154: WARNING - Another app already uses the domain cloud.maindomain.tld to serve a CalDAV/CardDAV feature. You may encounter issues when dealing with your calendar or address book.
2024-08-01 16:29:06,155: DEBUG - + ynh_print_log 'Another app already uses the domain cloud.maindomain.tld to serve a CalDAV/CardDAV feature. You may encounter issues when dealing with your calendar or address book.'
2024-08-01 16:29:06,155: DEBUG - + echo -e 'Another app already uses the domain cloud.maindomain.tld to serve a CalDAV/CardDAV feature. You may encounter issues when dealing with your calendar or address book.'
2024-08-01 16:29:06,155: DEBUG - + sed --in-place --regexp-extended '/location = \/\.well\-known\/(caldav|carddav)/d' ../conf/nginx.conf
2024-08-01 16:29:06,213: DEBUG - + ynh_add_nginx_config
2024-08-01 16:29:06,213: DEBUG - + local finalnginxconf=/etc/nginx/conf.d/cloud.maindomain.tld.d/nextcloud.conf
2024-08-01 16:29:06,213: DEBUG - + ynh_add_config --template=nginx.conf --destination=/etc/nginx/conf.d/cloud.maindomain.tld.d/nextcloud.conf
```
Although Nextcloud is the only app installed on this subdomain (and I don’t think I’ve another app capable of manage cardav/caldav stuffs.

File: scripts/_common.sh [hunk 2]
```diff
@@ -8,6 +8,19 @@
...
+    # wait for the nextcloud configuration file to disappear from nginx before checking the CalDAV/CardDAV URL.
+    timeout=30
+    for i in $(seq 1 $timeout); do
+        if ! ynh_exec_warn_less nginx -T | grep --quiet "# configuration file /etc/nginx/conf.d/$domain.d/$app.conf:"; then
+            break
+        fi
+        sleep 1
+    done
+}
+
```
Comment (kay0u) [inline scripts/_common.sh:22]: https://github.com/YunoHost/yunohost/blob/5d15c00d921927825a0bf98b0c5d872dac57d1b7/helpers/helpers.v1.d/utils#L112-L113

```suggestion
# Wait untils nginx has fully reloaded (avoid curl fail with http2) 
sleep 2
```

**Comment by kay0u** [comment]
!testme

**Review by ericgaspar — APPROVED** [review]

**Comment by nathanael-h** [comment]
Many many thanks for the removal of the chown on the data dir !!!! Again yesterday I faced a 2 hours upgrade of Nextcloud and was about to dig into this issue. Thank you, thank you, thank you !!! :hugs:

**Pull request merged**